### PR TITLE
check existence of machine config before further operations

### PR DESCRIPTION
**Pull request by HiteshRepo**
*Issue #, if available:*
#6039 

*Description of changes:*
check existence of machine config before further operations

### CODE REVIEW

**Comment by HiteshRepo** [comment]
@chrisdoherty4 could you please take a look at this?

**Comment by chrisdoherty4** [comment]
@HiteshRepo Apologies for the slowness in the last few hectic weeks. We'll take a look shortly. 

/cc chrisdoherty4 jacobweinstock

File: pkg/providers/tinkerbell/tinkerbell.go [hunk 1]
```diff
@@ -263,19 +263,23 @@ func (p *Provider) DatacenterConfig(_ *cluster.Spec) providers.DatacenterConfig
 func (p *Provider) MachineConfigs(_ *cluster.Spec) []providers.MachineConfig {
 	configs := make(map[string]providers.MachineConfig, len(p.machineConfigs))
 	controlPlaneMachineName := p.clusterConfig.Spec.ControlPlaneConfiguration.MachineGroupRef.Name
-	p.machineConfigs[controlPlaneMachineName].Annotations = map[string]string{p.clusterConfig.ControlPlaneAnnotation(): "true"}
-	if p.clusterConfig.IsManaged() {
-		p.machineConfigs[controlPlaneMachineName].SetManagedBy(p.clusterConfig.ManagedBy())
+	if _, ok := p.machineConfigs[controlPlaneMachineName]; ok {
```
Comment (chrisdoherty4) [inline pkg/providers/tinkerbell/tinkerbell.go:266]: I don't think we want to silently ignore the fact a machine config is missing. I think we can prevent this from ever happening by validating all machine configs are present in `NewProvider()`.

Specifically, on [L117](https://github.com/aws/eks-anywhere/blob/main/pkg/providers/tinkerbell/tinkerbell.go#L117) we `TinkebrellMachineConfig` spec but we don't handle it not being present.

**Review by chrisdoherty4 — CHANGES_REQUESTED** [review]
I don't think we want to silently ignore the fact a machine config is missing. I think we can prevent this from ever happening by validating all machine configs are present in NewProvider().

Specifically, on L117 we TinkebrellMachineConfig spec but we don't handle it not being present.

File: pkg/providers/tinkerbell/tinkerbell.go [hunk 2]
```diff
@@ -263,19 +270,23 @@ func (p *Provider) DatacenterConfig(_ *cluster.Spec) providers.DatacenterConfig
...
```
Comment (chrisdoherty4) [inline pkg/providers/tinkerbell/tinkerbell.go:273]: I think we can revert this change. If these machine configs are missing, there's a catastrophic problem and silently ignoring the fact may result in side effects with processes later on.

The checks added in `NewProvider` should be sufficient to protect against this issue now.

**Review by chrisdoherty4 — CHANGES_REQUESTED** [review]
@HiteshRepo thanks again. I've added 1 final comment regarding reverting the original changes.

**Comment by chrisdoherty4** [comment]
Looking great @HiteshRepo, thank you for your submissions.

**Review by chrisdoherty4 — APPROVED** [review]

**Pull request merged**